mail_tracking/static/src/js/failed_message/discuss.js

```diff
@@ -0,0 +1,398 @@
+                });
+                if (message) {
+                    message.setFailed(data.needs_actions);
+                    if (message.isFailed() === false) {
+                        self._removeMessageFromThread(
+                            'mailbox_failed', message);
+                    } else {
+                        self._addMessageToThreads(message, []);
+                        var channelFailed = self.getMailbox('failed');
+                        channelFailed.invalidateCaches();
+                    }
+                    self._mailBus.trigger('update_message', message, data.type);
+                }
+            });
+
+            if (data.needs_actions) {
+                // Increase failed counter if message is marked as failed
+                failed.incrementMailboxCounter(data.message_ids.length);
+            } else {
+                // Decrease failed counter if message is remove from failed
+                failed.decrementMailboxCounter(data.message_ids.length);
+            }
+
+            // Trigger event to refresh threads
+            this._mailBus.trigger('update_failed', failed.getMailboxCounter());
+        },
+    });
+
+    Discuss.include({
+        events: _.extend({}, Discuss.prototype.events, {
+            'click .o_failed_message_retry': '_onRetryFailedMessage',
+            'click .o_failed_message_reviewed': '_onMarkFailedMessageReviewed',
+        }),
+
+        /**
+        * Paramaters used to render 'failed' mailbox entry in Discuss
+        *
+        * @private
+        * @returns {Object}
+        */
+        _sidebarQWebParams: function () {
+            var failed = this.call('mail_service', 'getMailbox', 'failed');
+            return {
+                activeThreadID: this._thread ? this._thread.getID() : undefined,
+                failedCounter: failed.getMailboxCounter(),
+            };
+        },
+
+        /**
+        * Render 'failed' mailbox menu entry in Discuss
+        *
+        * @private
+        * @returns {jQueryElementt}
+        */
+        _renderSidebar: function () {
+            var $sidebar = this._super.apply(this, arguments);
+            // Because Odoo implementation isn't designed to be inherited
+            // properly, we inject 'failed' button using jQuery.
+            var $failed_item = $(QWeb.render('mail_tracking.SidebarFailed',
+                this._sidebarQWebParams()));
+            $failed_item.insertAfter(
+                $sidebar.find(".o_mail_discuss_title_main").filter(":last"));
+            return $sidebar;
+        },
+
+        /**
+         * Overrides to listen click on 'Set all as reviewed' button
+         *
+         * @Override
+         */
+        _renderButtons: function () {
+            this._super.apply(this, arguments);
+            this.$buttons
+                .on('click', '.o_mail_discuss_button_set_all_reviewed',
+                    this._onSetAllAsReviewedClicked.bind(this));
```

Comment:
It's better to lazily evaluate and bind the event. Otherwise, overriding it wouldn't have effect.

```suggestion
                    $.proxy(this, "_onSetAllAsReviewedClicked"));
```